src/thread_task.rs

```diff
@@ -834,163 +835,151 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
                             }
-                            let frame_hdr = &***(*f).frame_hdr.as_ref().unwrap();
+                            let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
                             let tile_row_base =
-                                frame_hdr.tiling.cols * (*f).frame_thread.next_tile_row[p as usize];
+                                frame_hdr.tiling.cols * f.frame_thread.next_tile_row[p as usize];
                             if p != 0 {
-                                let p1_0 =
-                                    (*f).frame_thread.entropy_progress.load(Ordering::SeqCst);
-                                if p1_0 < (*t).sby {
+                                let p1_0 = f.frame_thread.entropy_progress.load(Ordering::SeqCst);
+                                if p1_0 < t.sby {
                                     break 'next;
                                 }
-                                (*f).task_thread
+                                f.task_thread
                                     .error
                                     .fetch_or((p1_0 == TILE_ERROR) as c_int, Ordering::SeqCst);
                             }
-                            let frame_hdr = &***(*f).frame_hdr.as_ref().unwrap();
+                            let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
                             for tc_0 in 0..frame_hdr.tiling.cols {
-                                let ts: *mut Rav1dTileState = &mut *((*f).ts)
+                                let ts: *mut Rav1dTileState = &mut *(f.ts)
                                     .offset((tile_row_base + tc_0) as isize)
                                     as *mut Rav1dTileState;
                                 let p2 = ::core::intrinsics::atomic_load_seqcst(
                                     &mut *((*ts).progress).as_mut_ptr().offset(p as isize)
                                         as *mut atomic_int,
                                 );
-                                if p2 < (*t).recon_progress {
+                                if p2 < t.recon_progress {
                                     break 'next;
                                 }
-                                (*f).task_thread
+                                f.task_thread
                                     .error
                                     .fetch_or((p2 == TILE_ERROR) as c_int, Ordering::SeqCst);
                             }
-                            if ((*t).sby + 1) < (*f).sbh {
-                                let next_t: *mut Rav1dTask = &mut *t.offset(1) as *mut Rav1dTask;
-                                *next_t = (*t).clone();
+                            if (t.sby + 1) < f.sbh {
+                                // add sby+1 to list to replace this one
+                                let next_t: *mut Rav1dTask =
+                                    &mut *(t as *mut Rav1dTask).offset(1) as *mut Rav1dTask;
```

Comment:
Offsetting/slicing a reference seems kind of wrong?  Is it?

Reply:
It is, and we can't be leaving this forever. The reference here points into an array. We need to replace this with indexing later, but this isn't the place to do that.

Reply:
Okay, we don't need to fix it rn. But can you add a comment here and where `t` is declared saying it needs to be a slice eventually?